sympy/physics/control/lti.py

```diff
@@ -249,6 +252,147 @@ def expand(self):
+        >>> tf2.pole()
+        [-I, -I, I, I]
+        >>> tf3 = TransferFunction(s**2, a*s + p, s)
+        >>> tf3.pole()
+        [-p/a]
+
+        """
+        if self.den.is_number:
+            return []
+        p = Poly(self.den, self.var)
+        poles = None
+        try:
+            cmp_matrix = Matrix.companion(p)
+        except ValueError:
+            poles = roots(p, self.var)
+
+        if not poles:
+            try:
+                poles = cmp_matrix.eigenvals()
+            except MatrixError:
+                raise ValueError("Could not compute poles of this transfer function.")
+
+        rts = []
+        for key, val in poles.items():
+            rts.extend([key]*val)
+
+        return rts
+
+    def zero(self):
+        """
+        Returns the zeros of a transfer function.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> tf1 = TransferFunction((p + 3)*(p - 1), (p - 1)*(p + 5), p)
+        >>> tf1.zero()
+        [1, -3]
+        >>> tf2 = TransferFunction((1 - s)**2, (s**2 + 1)**2, s)
+        >>> tf2.zero()
+        [1, 1]
+        >>> tf3 = TransferFunction(s**2, a*s + p, s)
+        >>> tf3.zero()
+        [0, 0]
+
+        """
+        if self.num.is_number:
+            return []
+        p = Poly(self.num, self.var)
+        zeros = None
+        try:
+            cmp_matrix = Matrix.companion(p)
+        except ValueError:
+            zeros = roots(p, self.var)
+
+        if not zeros:
+            try:
+                zeros = cmp_matrix.eigenvals()
+            except MatrixError:
+                raise ValueError("Could not compute zeros of this transfer function.")
+
+        rts = []
+        for key, val in zeros.items():
+            rts.extend([key]*val)
+
+        return rts
+
+    @property
+    def is_stable(self):
+        """
+        Returns True if the transfer function is stable; else False.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p
+        >>> from sympy.core.symbol import symbols
+        >>> q, r = symbols('q, r', negative=True)
+        >>> from sympy.physics.control.lti import TransferFunction
+        >>> tf1 = TransferFunction((1 - s)**2, (s + 1)**2, s)
+        >>> tf1.is_stable
+        True
+        >>> tf2 = TransferFunction((1 - p)**2, (s**2 + 1)**2, s)
+        >>> tf2.is_stable
+        False
+        >>> tf3 = TransferFunction(4, q*s - r, s)
+        >>> tf3.is_stable
+        False
+
+        """
+        poles_ = sympify(self.pole())
+        if poles_ == []:
+            return True
+        else:
+            for pole in poles_:
+                p = pole.as_real_imag()[0]
+                if p.is_negative is None:
+                    raise ValueError("Could not check the stability of this transfer function.")
+                if not p.is_negative:
+                    return False
+            return True
```

Comment:
No need to special case there being no poles
```suggestion
        for pole in poles_:
            p = pole.as_real_imag()[0]
            if p.is_negative is None:
                raise ValueError("Could not check the stability of this transfer function.")
            if not p.is_negative:
                return False
        return True
```